### ISSUE TITLE
Wrong order status selected when order has no status

### ISSUE THREAD

**Issue by idnovate** [PrestaShop/PrestaShop#32743]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

When the order is not created successfully due to an error, the order is created without a status:
![image](https://github.com/PrestaShop/PrestaShop/assets/15104724/82f600f2-c4ae-491a-9dfc-217f7e33f90f)

The problem is that when you try to edit the order, the dropdown is selected with the first order status from the list.
![image](https://github.com/PrestaShop/PrestaShop/assets/15104724/141f6eb5-8d55-4884-b84a-3504120dbe1c)

### Expected behavior

The dropdown should be empty, as the dropdown from the orders list.

### Steps to reproduce

1. Generate an error 500 when creating the order. Edit the file `/classes/PaymentModule.php`, search for `$orderData = $this->createOrderFromCart` and place this code after:

```
header("HTTP/1.1 500 Internal Server Error");
trigger_error("Error 500", E_USER_ERROR);
die;
```

2. Create an order from the frontoffice. 
3. The order is created without status:
![image](https://github.com/PrestaShop/PrestaShop/assets/15104724/5aad3455-798e-46f9-87de-5f5d596d7b17)
4. Edit the order:
![image](https://github.com/PrestaShop/PrestaShop/assets/15104724/141f6eb5-8d55-4884-b84a-3504120dbe1c)

### PrestaShop version(s) where the bug happened

8.0.4

**Comment by AchrafKchaou1991** [comment]
Hello @idnovate 
I tried to reproduce the issues using PS 8.0 .4 but an exception appears :

[Issue 32743.webm](https://github.com/PrestaShop/PrestaShop/assets/131356155/e7fdeb24-404d-441c-87d8-b84d1103ef5f)

Can you check please ? Did I miss any steps?
Waiting for your feedback,
Thank you

**Comment by idnovate** [comment]
You have added the code

```
header("HTTP/1.1 500 Internal Server Error");
trigger_error("Error 500", E_USER_ERROR);
die;
```

As a parameter of the function.

I meant to add it after the function call. In your case, around line 359.

**Comment by AureRita** [comment]
Hi @idnovate 

Thank you for your report and for your details, I reproduce the issue with PrestaShop version 1.7.8.9,as you can see : 

[Untitled_ Jun 2, 2023 2_37 PM.webm](https://github.com/PrestaShop/PrestaShop/assets/100699445/3371cd02-9d37-4ffc-96c5-94323d04121f)

 I'll add this to the backlog so it can be fixed.

Please be aware that some issues might take a very long time to be resolved. 

Thanks

**Comment by idnovate** [comment]
_Bump to avoid expiration._

**Comment by paulnoelcholot** [comment]
Hello @idnovate 

Thank you for your feedback, could you please let us know if you are able to reproduce the bug on a maintained version (9.1.x and 9.2.x)?

Thanks!

**Comment by idnovate** [comment]
Tested in PS 9.1.1. Same error.

**Comment by ibahloul-ps** [comment]
Hello @idnovate 
Thanks for you report!

We think the solution should be to create a specific status if a payment module fails, so that the merchants is aware that there is a problem and can address it 👍

**Comment by boo-code** [comment]
Status check on 9.2, since this thread now has five PRs attached to it and they do not all do the same thing.

**The dropdown half is done.** #41950 merged on 2026-07-11 and is present on `9.2.x`:

```php
// src/PrestaShopBundle/Form/Admin/Sell/Order/UpdateOrderStatusType.php
$hasCurrentStatus = !empty($options['data']['new_order_status_id']);
...
'placeholder' => $hasCurrentStatus ? false : '',
```

with its unit test alongside it. So the symptom @idnovate opened this for — the dropdown pre-selecting the first status on a statusless order — no longer happens. What remains open is the cause: orders being left without a status in the first place.

**The two open PRs are complementary, not competing.**

- **#41697** (@idnovate) wraps the whole creation loop in a `try`, so any error mid-process still leaves the order with `PS_OS_ERROR` rather than `current_state = 0`. That is the general safety net.
- **#42056** (@paulschwahn) fixes one concrete way in. Verified on 9.2:

```
raw "<p></p>"   -> stripped ""   empty=true   Validate::isCleanHtml("") = true
Message::validateFields('')      -> 'Property Message->message is empty.'
```

The guard today is `if (!empty($message))` on the raw message and then `Validate::isCleanHtml($message)`. A message made only of markup passes both — it is not empty before stripping, and an empty string is clean HTML — so `Message::add()` is reached with nothing in it, throws, and the order never gets its status. #42056 adds `!empty($strippedMessage)`, which is exactly the missing condition.

So #41697 catches whatever else may go wrong, and #42056 removes a specific trigger that would otherwise keep firing. Both are worth having; neither makes the other redundant.

**One practical note on #42056.** Its blocking review is @jolelievre's `CHANGES_REQUESTED` from 2026-08-06, and it was purely a retarget request — "New features → develop, Bug fixes → 9.2.x". @paulschwahn rebased onto `9.2.x` the same day and said so on the thread. The review has just never been re-submitted, so the PR reads as blocked by something already done.

**And an ordering point for whoever merges them.** Both touch `validateOrder()` in `classes/PaymentModule.php`, and #41697's `try` wraps the region #42056 edits, so whichever lands second will need a rebase. They are on different branches at the moment as well — #41697 on `develop`, #42056 on `9.2.x`.

Nothing prepared from me here; between the merged dropdown fix and these two, the issue looks covered.
